test/test_serialization.h

```diff
@@ -529,7 +529,7 @@ TEST(serialization, sequential_model) {
   network<sequential> net1, net2;
 
   net1 << fully_connected_layer(10, 16) << tanh_layer(16)
-       << average_pooling_layer<relu>(4, 4, 1, 2)
+       << average_pooling_layer(4, 4, 1, 2) << relu_layer(2, 2, 1)
```

Comment:
The number of layers is increased to 4 from 3. You may update the below for loop condition as well.